wallet/test/Test/Pos/Util/MnemonicSpec.hs

```diff
@@ -77,6 +77,11 @@ spec = do
         it "CBackupPhrase from JSON" $ forM_ testVectors $ \(bytes, _, mnemonic, _, _) ->
             Aeson.decode (jsonV0Compat bytes) `shouldBe` pure (CBackupPhrase mnemonic)
 
+        it "CBackupPhrase from legacy JSON" $
+            (shouldBe @(Maybe (CBackupPhrase 12)))
+            (Aeson.decode legacy)
+            (Aeson.decode modern)
+
```

Comment:
Indeed. Since we already have the `testVectors`, I'd suggest to add an element to the tuple with the `legacy` version of it, and perform a test similar to the above:

```hs
it "CBackupPhrase from legacy JSON" $ forM_ testVectors $ \(_, legacyBytes_, mnemonic, _, _) ->
    Aeson.decode (jsonV0Compat legacyBytes) `shouldBe` pure (CBackupPhrase mnemonic)

it "Mnemonic from legacy JSON" $ forM_ testVectors $ \(_, legacyBytes_, mnemonic, _, _) ->
    Aeson.decode legacyBytes `shouldSatisfy` isLeft  
```

Reply:
Also, think that the parser use to accept any combination of space-separated words. So:

```
["abandon", "abandon abandon", "abandon abandon abandon abandon abandon", "abandon abandon abandon abandon"]

["abandon abandon abandon abandon abandon abandon abandon abandon abandon abandon abandon abandon "]
```

were both valid inputs. Try to reflect that aspect in the test vectors you add :+1: